sci-mathematics/octave/octave-9.1.0.ebuild

```diff
@@ -0,0 +1,273 @@
+# flags whose existing descriptions conflict with the obvious way we
+# would want to use them in octave. In any case, upstream doesn't really
+# support imagemagick, only graphicsmagick (bug 864785).
+RDEPEND="
+	app-arch/bzip2
+	app-arch/unzip
+	app-arch/zip
+	app-text/ghostscript-gpl
+	sys-apps/texinfo
+	dev-libs/libpcre2
+	sys-libs/ncurses:=
+	sys-libs/zlib
+	virtual/blas
+	virtual/lapack
+	curl? ( net-misc/curl:= )
+	fftw? ( sci-libs/fftw:3.0= )
+	fltk? ( >=x11-libs/fltk-1.3:1=[opengl,xft] )
+	glpk? ( sci-mathematics/glpk:= )
+	gnuplot? ( sci-visualization/gnuplot )
+	hdf5? ( sci-libs/hdf5:= )
+	imagemagick? ( media-gfx/graphicsmagick:=[cxx] )
+	java? ( >=virtual/jre-1.8:* )
+	json? ( dev-libs/rapidjson )
+	klu? ( sci-libs/klu:= )
+	opengl? (
+		media-libs/freetype:=
+		media-libs/fontconfig:=
+		virtual/glu
+		x11-libs/gl2ps:=
+	)
+	portaudio? ( media-libs/portaudio )
+	postscript? (
+		app-text/epstool
+		media-gfx/pstoedit
+		>=media-gfx/fig2dev-3.2.9-r1
+	)
+	gui? (
+		qt5? (
+			dev-qt/qtcore:5
+			dev-qt/qtgui:5
+			dev-qt/qthelp:5
+			dev-qt/qtnetwork:5
+			dev-qt/qtopengl:5
+			dev-qt/qtprintsupport:5
+			dev-qt/qtwidgets:5
+			x11-libs/qscintilla:=[qt5]
+		)
+		qt6? (
+			dev-qt/qtbase:6[gui,opengl,network,widgets]
+			x11-libs/qscintilla:=[qt6]
+		)
+	)
+	qhull? ( media-libs/qhull:= )
+	qrupdate? ( sci-libs/qrupdate:= )
+	readline? ( sys-libs/readline:= )
+	sndfile? ( media-libs/libsndfile )
+	sparse? (
+		sci-libs/arpack:=
+		sci-libs/camd:=
+		sci-libs/ccolamd:=
+		sci-libs/cholmod:=
+		sci-libs/colamd:=
+		sci-libs/cxsparse:=
+		sci-libs/umfpack:=
+	)
+	spqr? ( sci-libs/spqr:= )
+	ssl? (
+		dev-libs/openssl:=
+	)
+	sundials? (
+		klu? ( >=sci-libs/sundials-4:=[sparse] )
+		!klu? ( >=sci-libs/sundials-4:= )
+	)
+	X? ( x11-libs/libX11:= )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-util/gperf
+	virtual/pkgconfig
+	doc? (
+		dev-texlive/texlive-fontsrecommended
+		dev-texlive/texlive-plaingeneric
+		dev-texlive/texlive-metapost
+		virtual/latex-base
+	)
+	java? ( >=virtual/jdk-1.8:* )
+	qt5? ( dev-qt/linguist-tools:5 )
+	qrupdate? ( app-misc/pax-utils )
+	sparse? ( app-misc/pax-utils )
+"
+
+# There are three ways to plot in Octave:
+#
+#   1. The old gnuplot renderer
+#   2. The OpenGL renderer using the FLTK backend
+#   3. The OpenGL renderer using the Qt backend
+#
+# It's possible to use the Qt GUI without OpenGL, but OpenGL rendering
+# is all that FLTK is used for, so it doesn't make sense to enable
+# USE=fltk without USE=opengl.
+#
+# Building without either USE=gnuplot or USE=opengl is technically legal,
+# but will leave you unable to plot anything.
+#
+# Octave's FLTK support is unofficially deprecated, in the sense that
+# you'll often get "why are you using FLTK?" in response to
+# bugs. (Upstream bug 59321 for a random example.) In the future, it
+# will probably make sense to merge USE=opengl and USE=X into USE=qt[56],
+# dropping USE=fltk entirely.
+REQUIRED_USE="
+	?? ( qt5 qt6 )
```

Comment:
Per [Qt packaging policy](https://wiki.gentoo.org/wiki/Project:Qt/Policies) we should not introduce the `qt5` flag and `REQUIRED_USE` here.

Instead we should add only the `qt6` flag, if enabled gui is built with qt6, otherwise it is built with qt5.